parcel.py

```diff
@@ -359,96 +578,128 @@ def parcel(dt=.1, z_max=200., w=1., T_0=300., p_0=101300.,
+
   if opts["chem_dsl"] or opts["chem_dsc"] or opts["chem_rct"]:
     for key in _Chem_g_id.keys():
       state.update({ key : np.array([opts[key]])})
 
   info = { "RH_max" : 0, "libcloud_Git_revision" : libcloud_version,
            "parcel_Git_revision" : parcel_version }
 
-  micro = _micro_init(aerosol, opts, state, info)
-
-  with _output_init(micro, opts, spectra) as fout:
-    # adding chem state vars
-    if micro.opts_init.chem_switch:
-      state.update({ "SO2_a" : 0.,"O3_a" : 0.,"H2O2_a" : 0.,})
-      state.update({ "CO2_a" : 0.,"HNO3_a" : 0.})
-
-      micro.diag_all() # selecting all particles
-      micro.diag_chem(_Chem_a_id["NH3_a"])
-      state.update({"NH3_a": np.frombuffer(micro.outbuf())[0]})
-
-    # t=0 : init & save
-    _output(fout, opts, micro, state, 0, spectra)
-
-    # timestepping
-    for it in range(1,nt+1):
-      # diagnostics
-      # the reasons to use analytic solution:
-      # - independent of dt
-      # - same as in 2D kinematic model
-      state["z"] += w * dt
-      state["t"] = it * dt
-
-      # pressure
-      if pprof == "pprof_const_th_rv":
-        # as in icicle model
-        p_hydro = _p_hydro_const_th_rv(state["z"], p_0, th_0, r_0)
-      elif pprof == "pprof_const_rhod":
-        # as in Grabowski and Wang 2009
-        rho = 1.13 # kg/m3  1.13
-        state["p"] = _p_hydro_const_rho(state["z"], p_0, rho)
-
-      elif pprof == "pprof_piecewise_const_rhod":
-        # as in Grabowski and Wang 2009 but calculating pressure
-        # for rho piecewise constant per each time step
-        state["p"] = _p_hydro_const_rho(w*dt, state["p"], state["rhod"][0])
-
-      else: raise Exception("pprof should be pprof_const_th_rv, pprof_const_rhod, or pprof_piecewise_const_rhod")
-
-      # dry air density
-      if pprof == "pprof_const_th_rv":
-        state["rhod"][0] = common.rhod(p_hydro, th_0, r_0)
-        state["p"] = common.p(
-          state["rhod"][0],
-          state["r_v"][0],
-          common.T(state["th_d"][0], state["rhod"][0])
-        )
-
-      else:
-        state["rhod"][0] = common.rhod(
-          state["p"],
-          common.th_dry2std(state["th_d"][0], state["r_v"][0]),
-          state["r_v"][0]
-        )
-
-      # microphysics
-      _micro_step(micro, state, info, opts, it, fout)
-
-      # TODO: only if user wants to stop @ RH_max
-      #if (state["RH"] < info["RH_max"]): break
-
-      # output
-      if (it % outfreq == 0):
-        print(str(round(it / (nt * 1.) * 100, 2)) + " %")
-        rec = it/outfreq
-        _output(fout, opts, micro, state, rec, spectra)
-
-    _save_attrs(fout, info)
-    _save_attrs(fout, opts)
-
-    if wait != 0:
-      for it in range (nt+1, nt+wait):
+  # Initialize the selected scheme
+  if scheme == "lgrngn":
+      micro = _micro_init(aerosol, opts, state, info)
+      fout = _output_init(micro, opts, spectra)
+  elif scheme == "blk_1m":
+      micro = _micro_init_blk_1m(opts, state, info)
```

Comment:
returned variable is an opts structure, not a particles_t structure as in lgrngn. I would rename it to avoid confusion (same for blk_ice)